### ISSUE TITLE
Simple buying strategies 🎯

### ISSUE THREAD

**Issue by pahautelman** [pahautelman/bitcoin-bot#3]
- [ ] Implement other buying strategies based on indicators

Indicators to consider:
- trend:
    - 50 and 200 exponential moving average (EMA; lagging trend indicator), used to identify the direction of the trend
    - average directional index (ADX, lagging), measures the strength of the trend
    - they are both similar, ADX seems to be more expressive. It indicates both direction and strength of trend (https://www.investopedia.com/articles/trading/07/adx-trend-indicator.asp). You can use both, but the DEMA cross-over signal strength might be hard to calculate. 
- momentum:
    - RSI (leading indicator), indicates overbought/oversold conditions
    - stochastic oscillator (leading indicator), measures the momentum and speed of a price movement
    - MACD (lagging indicator), tool to examine rapid price change. Histogram peak indicates buy/sell signal.
    - SO and MACD capture potential crossover, while RSI does not. SO and RSI easier to calculate signal strength. 
- oscillators:
    - commodity channel index (CCI), identifies overbought/oversold conditions
    - William %R, measures overbought or oversold levels
    - both similar to SO and RSI, can be ignored
- volatility:
    - Bollinger bands (BB, lagging mean reversion indicator), shows overbought/oversold conditions and volatility
    - average true range (ATR), measure market volatility
    - quite different in information. Both can be implemented
- volume:
    - ob-balance volume (OBV), relates price to volume and helps identify accumulation or distribution
    - Chaikin money flow (CMF), measures the strength of money flow into or out of a security
    - Visible Range Volume Profile
- sentiment:
    - fear and greed index
- support and resistance indicators:
    - fibonacci retracements, highlights potential reversal levels based on Fibonacci ratios
- pattern recognition:
    - recognize candlestick patterns to reveal potential reversals or continuations.
- *moon phase* indicator (leading trend indicator). Full moon signals a buy signal, short on naked moon.               

Strategies:
1. Moving average crossover strategy  
Indicators: SMA and EMA  
Strategy: Buy when the short-term EMA crosses above the longer-term SMA, indicating a potential uptrend. Sell when the EMA crosses below the SMA, signaling a potential downtrend  
Risk management: Implement stop-loss and take-profit levels to manage risk. Avoid trading during periods of reduced volatility

2. ~~RSI overbought/oversold strategy  
Indicators: RSI  
Strategy: Buy when RSI crosses below 30, indicating oversold conditions, and sell when RSI crosses above 70. This strategy looks for trend reversals at extreme RSI levels~~

3. ~~MACD crossover strategy  
Indicators: MACD  
Strategy: Buy when the MACD line crosses above the signal line and sell when it crosses below. Strategy aims to capture momentum shifts in the market.  
Risk management: Be cautious of false signals during sideways markets.~~

4. ~~BB reversion to mean 
Indicators: BB  
Strategy: Buy when the price touches or falls below the lower BB and then rises back inside the bands, indicating potential support. Sell when the price touches or exceeds the upper BB and then drops back inside, suggesting potential resistance.~~  

5. Ichimoku cloud breakout strategy  
Indicators: Ichimoku Cloud  
Strategy: Buy when the price crosses above the cloud, and the cloud is green (indicating an uptrend). Sell when the price crosses below the cloud, and the cloud is red.  
Risk management: Use the cloud as a dynamic stop-loss level. Adjust stop-loss levels as the cloud moves with the trend ???

6. Dual moving average crossover with RSA confirmation  
Indicators: SMA, RSI  
Strategy: Buy when the shorter-term SMA crosses above the longer-term SMA and the RSI is above a certain threshold (ie: 50). Sell when the shorter-term SMA crosses below the longer-term SMA or when the RSI falls bellow the threshold.
  

- [ ] Implement different investing strategies (for determining the size of each position).

It might be useful to split responsibilities of investor agent as:
* buying strategy (manages stop-losses, decides amount of USD portfolio to invest)
* selling strategy (decides amount of coin assets to sell)
  
Resources:
* https://medium.com/@ctjong27/implementing-trading-bot-strategies-b7ce28baacfe
* https://www.cryptohopper.com/blog/top-six-crypto-trading-indicators-and-how-to-use-them-on-cryptohopper-8204
* https://www.youtube.com/watch?v=sbKTRVWppZY

Position sizing strategies:
- Fixed percentage of portfolio: Allocate a fixed percentage of your total portfolio to each trade. For example, you might decide to invest 2% of your portfolio in each trade.
- Indicator strenght-based: The strength of the indicator can determine the size of the position. For example, a strong buy signal may prompt you to allocate a larger percentage of your portfolio (2%), while a weaker signal might result in a smaller allocation (1%).
- Stop-loss level based on the risk tolerance. The stop-loss level can be a percentage of the entry price (eg, 2% stop-loss).
- Ensemble of models to decide the size of the position. Use all/subset of buying strategies (eg: weighted average, based on the performance of each indicator) to decide the strength of the buy/sell signal
- risk-adjusted sizing. Position based on the risk of the trade. Calculate distance between the entry_price (EP) and stop-loss level (SL) and adjust sizing accordingly.
    - stop-loss = 0.1
    - EP = 28_000
    - => SL = 28_000 - 0.1 * 2_800 = 25_200
    - => potential loss = 2_800
    - risk tolerance = 0.05
    - => position sizing = 2_800 / 0.05 = 56_000
    - ??? 
- Percentage of ATR (Average True Range)
